trending_data/src/main/webapp/unique_window_handler.js

```diff
@@ -148,9 +148,10 @@ export class UniqueWindowHandler {
     }
     this.lastCode = this.marker.countryCode;
     this.initDataWindow();
-    await this.loadYTCategories();
-    await this.loadYTData(this.defaultYTCategory);
-    await this.loadTwitterData();
+    // execute data fetches in parallel
+    await Promise.all([this.loadYTCategories(),
```

Comment:
As Henry said this will fail if any of the requests fail - I think this is something to keep in mind for the future, but not necessary to fix now since some of the async functions are not handling errors gracefully either. For reference, ccording to this:
https://stackoverflow.com/questions/42158853/how-run-async-await-in-parallel-in-javascript/42158854#42158854

We could do something like the following:
```
const youtubeCategoryLoadResult = this.loadYTCategories();
const youtubeDataLoadResult = this.loadYTData();
const twitterDataLoadResult = this.loadTwitterData();
await youtubeCategoryLoadResult;
await youtubeDataLoadResult;
await twitterDataLoadResult;
```

we can then check for errors, etc. but at the same time have all 3 functions execute in parallel.